src/render.ts

```diff
@@ -14,6 +14,7 @@
  * @group render_utils
  */
 export function heading(text: string, level: number): string {
```

Comment:
Maybe we should restrict the type for level as well?

Reply:
When it is `not a number`, do you expect to throw an error or a warning and return the original value?

Reply:
Yes, it's done.

Reply:
I mean

```suggestion
export function heading(text: string, level: 1|2|3|4|5|6): string {
```

Reply:
I won't pin all my hopes on typescript, I will if you want.

Reply:
we can also use `|(number & {}` to make your life better? :p

Reply:
I still want to use the `number` type and handle the `level` inside the function, So that it can be called recursively without error.